accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
+- The extension application `E(x)` would be valid (not a compile-time error) where `x` is a fresh variable with static type *S* (to avoid type inference for any type parameters of `E` from affecting the already determined static type of `e`).
+- and `E` declares an instance member with the basename *m*.
 
-will fail. While we could make it work by inferring `T` as `Object`, we don't. We infer `T` *only* based on the receiver type, and therefore `T` is `String`, and `o` is not a valid argument (at least not when we remove implicit downcasts).
+Notice that the context type of the invocation does not affect whether the extension applies, and neither the context type nor the method invocation affects the type inference of `e`, but if the extension method itself is generic, the context type may affect the member invocation.
 
-### Extension Conflict Resolution
+##### Specificity
 
-If more than one extension applies to a specific member invocation, then we resort to a heuristic to choose one of the extensions to apply. If exactly one of them is "more specific" than all the others, that one is chosen. Otherwise it is a compile-time error.
+When more than one extension is accessible and applicable to a member invocation, we define a partial ordering on those extensions wrt. that member invocation, so that we can choose the "best" candidate which will have its extension member be implicitly invoked.
 
-An extension with `on` type clause *T*<sub>1</sub> is more specific than another extension with `on` type clause *T*<sub>2</sub> iff 
+Let *i* be a member invocation with target expression `e` and corresponding member name *m*, and let `E1` and `E2` denote different accessible and applicable extensions for *i*. Let *T<sub>1</sub>* be the instantiated "on" type of `E1` wrt. `e` and let *T<sub>2</sub>* be the instantiated "on" type of `E2` wrt. `e`. Then `E1` is more specific than `E2` wrt. *i* iff:
 
-1. The latter extension is declared in a platform library, and the former extension is not, or
-2. they are both declared in platform libraries or both declared in non-platform libraries, and
-3. the instantiated type (the type after applying type inference from the receiver) of *T*<sub>1</sub> is a subtype of the instantiated type of *T*<sub>2</sub> and either
+1. The `E2` extension is declared in a platform library and the `E1` extension is not, or
+2. either both or neither are declared in platform libraries and
+3. *T<sub>1</sub>* is a subtype of of *T<sub>2</sub>* and either
 4. not vice versa, or
-5. the instantiate-to-bounds type of *T*<sub>1</sub> is a subtype of the instantiate-to-bounds type of *T*<sub>2</sub> and not vice versa.
+5. the instantiate-to-bounds `on` type of `E1` is a subtype of the instantiate-to-bounds `on` type of `E2` and not vice versa.
```

Comment:
Instantiation to bounds is not defined if `E1`, `E2` is not in scope.

Reply:
`E1` denotes the extension, not the name of the extension, and instantiate-to-bounds type is defined on the extension declaration itself, so yes it is defined :)